### PR TITLE
util: make page-server IPv6 safe

### PR DESCRIPTION
**Pull request by adrianreber**
The function run_tcp_server() was the last place CRIU was still using the IPv4 only function inet_ntoa(). It was only used during a print, so that it did not really break anything, but with this commit the output is now no longer:

` Accepted connection from 0.0.0.0:58396`

but correctly displaying the IPv6 address

` Accepted connection from ::1:58398`

if connecting via IPv6.

### CODE REVIEW

**Review by rst0git — APPROVED** [review]
LGTM

File: criu/util.c [hunk 1]
```diff
@@ -1139,9 +1139,17 @@ int run_tcp_server(bool daemon_mode, int *ask, int cfd, int sk)
...
 			goto err;
-		} else
-			pr_info("Accepted connection from %s:%u\n", inet_ntoa(caddr.sin_addr),
-				(int)ntohs(caddr.sin_port));
+		} else {
+			char port[6];
+			char address[INET6_ADDRSTRLEN];
+			ret = getnameinfo((struct sockaddr *)&caddr, clen, address, sizeof(address), port, sizeof(port),
+					  NI_NUMERICHOST | NI_NUMERICSERV);
+			if (unlikely(ret)) {
```
Comment (avagin) [inline criu/util.c:1147]: nit: you don't need this unlikely here

File: criu/util.c [hunk 2]
```diff
@@ -1139,9 +1139,17 @@ int run_tcp_server(bool daemon_mode, int *ask, int cfd, int sk)
 		if (*ask < 0) {
 			pr_perror("Can't accept connection to server");
 			goto err;
-		} else
-			pr_info("Accepted connection from %s:%u\n", inet_ntoa(caddr.sin_addr),
-				(int)ntohs(caddr.sin_port));
+		} else {
```
Comment (avagin) [inline criu/util.c:1142]: else isn't needed here.

**Comment by adrianreber** [comment]
@avagin thanks for the review. I fixed both review comments and force pushed.

**Pull request merged**